aptos-move/framework/aptos-framework/sources/configs/Stake.move

```diff
@@ -828,18 +932,33 @@ module AptosFramework::Stake {
         let i = 0;
         let len = Vector::length(&validator_set.active_validators);
         let active_validators = Vector::empty();
-        validator_perf.num_blocks = 1;
+        validator_perf.num_blocks = 0;
         validator_perf.missed_votes = Vector::empty();
+
+        // Reset validator set's voting power tracking.
+        let validator_set_voting_power = borrow_global_mut<ValidatorSetVotingPower>(@CoreResources);
+        validator_set_voting_power.total = 0;
+        validator_set_voting_power.joining = 0;
+        validator_set_voting_power.withdrawing = 0;
+
+        // Process validators in the (new) active validator set.
         while (i < len) {
             let old_validator_info = Vector::borrow_mut(&mut validator_set.active_validators, i);
             let pool_address = old_validator_info.addr;
             let validator_config = borrow_global_mut<ValidatorConfig>(pool_address);
             let new_validator_info = generate_validator_info(pool_address, *validator_config);
-            if (new_validator_info.voting_power >= validator_set_config.minimum_stake &&
-                new_validator_info.voting_power <= validator_set_config.maximum_stake
-            ) {
+
+            // Restrict a validator's voting power to the max stake allowed.
+            if (new_validator_info.voting_power > validator_set_config.maximum_stake) {
+                new_validator_info.voting_power = validator_set_config.maximum_stake;
+            };
+
+            // A validator needs at least the min stake required to join the validator set.
+            if (new_validator_info.voting_power >= validator_set_config.minimum_stake) {
```

Comment:
this is another edge case not sure if we want to handle - if someone withdraw only 10 but resulted in the node below the minimum_stake, the actual reduction is going to be the all the voting power from that pool.

Reply:
the easiest thing to do is probably just prevent people from withdrawing below minimum_stake. they have to leave first

Reply:
Updated to prevent active validators (not considering pending_inactive) from unlocking if their remaining stake is less than min stake required. I also added a unit test just for that case.